### PR TITLE
Blockly no upstream diffs

### PR DESCRIPTION
**Pull request by jimmo**
Extra overrides and functionality patching in our blockly-blocks repo that allow us to use unmodified blockly (this is currently against blockly at [`701b187640adb3bfc757a327159d269a0b56e169`](https://github.com/groklearning/blockly/tree/701b187640adb3bfc757a327159d269a0b56e169)).

Also adds a build.sh that builds a single minified js for all blockly and blockly-blocks JS. (Optionally in whitespace only mode for fast local goodness)

The overall approach is:
 - Fix goog.require/goog.provide for all our files
 - Where possible, override the existing blockly functions by calling the original one and modifying the result (or other side effects).
 - Confine all the really bad overrides to core/blockly.js.

Also adds the event handler blocks for JS.

### CODE REVIEW

**Comment by timdawborn** [comment]
How are you expecting this to be tested since no branch of `groklearning/blockly` is currently at 701b187640adb3bfc757a327159d269a0b56e169 — manually checkout this commit and run against it and trust that this commit will be what's force pushed back to `develop-grok`?

**Comment by jimmo** [comment]
Yes, my exact plan was

```
$ git co develop-grok
$ git co -b develop-grok-archive
$ git push -u origin develop-grok-archive
$ git co develop-grok
$ git fetch upstream
$ git reset --hard <that commit>
$ git push -f
```

**Comment by jimmo** [comment]
But for testing -- locally that's fine to just checkout at the commit hash, if we want to test on staging we could do that too (or make another branch or tag at that commit and modify hte fabfile for that).

**Comment by jimmo** [comment]
I forgot to mention -- this is of course paired with https://github.com/groklearning/app/compare/blockly-no-upstream-diffs

File: .gitignore [hunk 1]
```diff
@@ -1 +1 @@
-.*.sw[opq]
+build/
```
Comment (timdawborn) [inline .gitignore]: Can you add the vim swapfile ignore back in? It's super useful.

File: blocks/microbit.js [hunk 2]
```diff
@@ -403,7 +404,7 @@ Blockly.Extensions.register('microbit_image_dropdown', function() {
     var field = self.getField('CATEGORY');
```
Comment (timdawborn) [inline blocks/microbit.js]: I can't tell whether or not it's an artefact of this change or an artefact of merging upstream, but the behaviour of the 2nd dropdown has changed between what's on prod and what's here. The UI element seems to be behaving differently. I'm guessing it's an upstream change, but it's super annoying.

**Pull request merged**